tests/integration/activities.test.js

```diff
@@ -167,18 +167,25 @@ describe('ACTIVITIES ROUTES - POST /activities/{_id}/card', () => {
       authToken = await getToken('training_organisation_manager');
     });
 
-    it('should create card', async () => {
+    it('should create card with valid form (testing template "survey" specifically)', async () => {
       const response = await app.inject({
         method: 'POST',
         url: `/activities/${activitiesList[0]._id}/cards`,
-        payload,
+        payload: { template: SURVEY },
         headers: { Cookie: `alenvi_token=${authToken}` },
       });
 
       const activityUpdated = await Activity.findById(activitiesList[0]._id).lean();
-
       expect(response.statusCode).toBe(200);
       expect(activityUpdated.cards.length).toEqual(5);
+
+      const newCardWithValidForm = await Card.countDocuments({
+        _id: activityUpdated.cards.slice(-1)[0],
+        template: 'survey',
+        'label.right': { $exists: true },
```

Comment:
Sur ma machine et celle d'Ulysse il y a un pb avec `$exists`. 

la condition `$exists: true` est OK dans tous les cas. Qua la clé existe belle et bien ou non. (Et ce avec des sous-champs comme avec des champs a la racine).
Ce test ne sert donc a rien.

En revanche `$exists: false``fonctionne comme on s'y attend. 

Tu peux tester avec ton ordi ? SI tu confirmes le bug, est-ce au'on le trqite dans ce ticket ou est-ce qu'on le fait dans un autre ?

Reply:
J'ai fait les tests suivants : 
je mets if (!this.label) this.label = {}; dans le modèle
si je teste avec label: { right: { $eq: '' } }  / label: { right: { $exists: true } } ça casse
si je teste avec 'label.right': { $exists: false } ça fonctionne
si je teste avec label: { riglmsldfht: { $eq: '' } } ça fonctionne
je mets if (!this.label) this.label = { right: '', left: '' }; dans le modèle
si je teste avec label: { right: { $eq: '' } }  / label: { right: { $exists: true } } ça fonctionne
si je teste avec 'label.right': { $exists: false } ça casse
si je teste avec label: { riglmsldfht: { $eq: '' } } ça fonctionne

Mon hypothese : si tu passes une clé qui n'est pas dans le modele alors elle n'est pas interprété dans la requête mongoose

Le test 'label.right': { $exists: true } plante si le prehook n'est pas bien défini (correctif de la PR) donc pour moi ton test sers bien a quelque chose

Qu'en penses tu ?

Reply:
@ulysseferreira pour info

Reply:
vu ensemble en visio